src/ngx_http_auth_jwt_module.c

```diff
@@ -20,543 +20,638 @@
+		 ngx_conf_set_flag_slot,
+		 NGX_HTTP_LOC_CONF_OFFSET,
+		 offsetof(auth_jwt_conf_t, redirect),
+		 NULL},
+
+		{ngx_string("auth_jwt_validation_type"),
+		 NGX_HTTP_MAIN_CONF | NGX_HTTP_SRV_CONF | NGX_HTTP_LOC_CONF | NGX_CONF_TAKE1,
+		 ngx_conf_set_str_slot,
+		 NGX_HTTP_LOC_CONF_OFFSET,
+		 offsetof(auth_jwt_conf_t, validation_type),
+		 NULL},
+
+		{ngx_string("auth_jwt_algorithm"),
+		 NGX_HTTP_MAIN_CONF | NGX_HTTP_SRV_CONF | NGX_HTTP_LOC_CONF | NGX_CONF_TAKE1,
+		 ngx_conf_set_str_slot,
+		 NGX_HTTP_LOC_CONF_OFFSET,
+		 offsetof(auth_jwt_conf_t, algorithm),
+		 NULL},
+
+		{ngx_string("auth_jwt_validate_sub"),
+		 NGX_HTTP_MAIN_CONF | NGX_HTTP_SRV_CONF | NGX_HTTP_LOC_CONF | NGX_CONF_FLAG,
+		 ngx_conf_set_flag_slot,
+		 NGX_HTTP_LOC_CONF_OFFSET,
+		 offsetof(auth_jwt_conf_t, validate_sub),
+		 NULL},
+
+		{ngx_string("auth_jwt_extract_request_claims"),
+		 NGX_HTTP_MAIN_CONF | NGX_HTTP_SRV_CONF | NGX_HTTP_LOC_CONF | NGX_CONF_1MORE,
+		 merge_extract_request_claims,
+		 NGX_HTTP_LOC_CONF_OFFSET,
+		 offsetof(auth_jwt_conf_t, extract_request_claims),
+		 NULL},
+
+		{ngx_string("auth_jwt_extract_response_claims"),
+		 NGX_HTTP_MAIN_CONF | NGX_HTTP_SRV_CONF | NGX_HTTP_LOC_CONF | NGX_CONF_1MORE,
+		 merge_extract_response_claims,
+		 NGX_HTTP_LOC_CONF_OFFSET,
+		 offsetof(auth_jwt_conf_t, extract_response_claims),
+		 NULL},
+
+		{ngx_string("auth_jwt_keyfile_path"),
+		 NGX_HTTP_MAIN_CONF | NGX_HTTP_SRV_CONF | NGX_HTTP_LOC_CONF | NGX_CONF_TAKE1,
+		 ngx_conf_set_str_slot,
+		 NGX_HTTP_LOC_CONF_OFFSET,
+		 offsetof(auth_jwt_conf_t, keyfile_path),
+		 NULL},
+
+		{ngx_string("auth_jwt_use_keyfile"),
+		 NGX_HTTP_MAIN_CONF | NGX_HTTP_SRV_CONF | NGX_HTTP_LOC_CONF | NGX_CONF_FLAG,
+		 ngx_conf_set_flag_slot,
+		 NGX_HTTP_LOC_CONF_OFFSET,
+		 offsetof(auth_jwt_conf_t, use_keyfile),
+		 NULL},
+
+		ngx_null_command};
+
+static ngx_http_module_t auth_jwt_context = {
+		NULL,				 /* preconfiguration */
+		init,				 /* postconfiguration */
+		NULL,				 /* create main configuration */
+		NULL,				 /* init main configuration */
+		NULL,				 /* create server configuration */
+		NULL,				 /* merge server configuration */
+		create_conf, /* create location configuration */
+		merge_conf	 /* merge location configuration */
 };
 
-
 ngx_module_t ngx_http_auth_jwt_module = {
-	NGX_MODULE_V1,
-	&ngx_http_auth_jwt_module_ctx,     /* module context */
-	ngx_http_auth_jwt_commands,        /* module directives */
-	NGX_HTTP_MODULE,                   /* module type */
-	NULL,                              /* init master */
-	NULL,                              /* init module */
-	NULL,                              /* init process */
-	NULL,                              /* init thread */
-	NULL,                              /* exit thread */
-	NULL,                              /* exit process */
-	NULL,                              /* exit master */
-	NGX_MODULE_V1_PADDING
-};
-
-
-static ngx_int_t ngx_http_auth_jwt_handler(ngx_http_request_t *r)
+		NGX_MODULE_V1,
```

Comment:
Here too

Reply:
Same here, it's just where it wants to be, I guess.